### DISCUSSION TITLE
Define elements using matrix

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#620] [Python AT]
Hi everyone,

I had a lattice sequence file which i wanted to use in PyAT, i converted the seq file to .mat using xsequence tool, but i faced problem with the wigglers elements conversion so i changed it in the sequence to drifts, now i need to insert the wigglers again in PyAT 

The wiggler is defined in the sequence file as : 

wiggler: matrix,l:= 4,rm11:= 1.000449786249,rm12:= 4.000639844715,rm21:= 0.00022577719708,rm22:= 1.00045326313,rm33:= 0.9786569514876,rm34:= 3.970145599493,rm43:=-0.01063922161568,rm44:= 0.97864807445;

I am asking how can i define elements in pyat using the transfer matrix ?

Thank you in advance

**Comment by swhite2401** [comment]
Please take a look at:
https://atcollab.github.io/at/p/api/at.lattice.elements.html#at.lattice.elements.M66

You will have to extend your 4x4 matrix to 6D.
